pkg/execution/state/redis_state/lua/saveResponse.lua

```diff
@@ -17,7 +17,11 @@ local stepID  = ARGV[1]
 local data    = ARGV[2]
 
 if redis.call("HEXISTS", keyStep, stepID) == 1 then
-	return -1
+  local existingData = cjson.decode(redis.call("HGET", keyStep, stepID))
```

Comment:
We need to move this to metadata as per that convo - no mutations in step state! :)

Reply:
- #1971